ui/src/components/settings/sub_pages/profile/mod.rs

```diff
@@ -0,0 +1,182 @@
+                ))
+                show_edit_fields.then(|| 
+                    {
+                    let new_username_val = use_ref(&cx, String::new);
+                    let new_status_message_val = use_ref(&cx, || format!("{}", status_message));
+                    rsx!(
+                    div {
+                        class: "edit-button", 
+                        Button {
+                            text: get_local_text("settings-profile.save-button"),
+                            onpress: move |_| {
+                                let new_username = new_username_val.with(|i| i.clone());
+                                if !new_username.is_empty() && new_username.len() > 3 {
+                                    username.set(new_username);
+                                }
+                                let new_status_message = new_status_message_val.with(|i| i.clone());
+                                status_message.set(new_status_message);
+                                edit_mode.set(!edit_mode);
+                            },
+                        },
+                    },
+                    div {
+                        class: "username", 
+                        Input {
+                            id: "username_text_field".to_owned(),
+                            focus: true,
+                            placeholder: format!("{}", username),
+                            disabled: false,
+                            onchange: move |value| {
+                                *new_username_val.write_silent() = value;
+                            }, 
+                            options: Options {
+                                with_clear_btn: true,
+                                ..Options::default()
+                            },
+                        },
+                    },
+                  div {
+                        class: "status-message-edit", 
+                        Input {
+                            placeholder: format!("{}", status_message),
+                            disabled: false,
+                            onchange: move |value| {
+                                *new_status_message_val.write_silent() = value;
+                            }, 
+                            options: Options {
+                                with_clear_btn: true,
+                                ..Options::default()
+                            }
+                        }
+                    },
+                    )}),
+              
+            },
+        }
+    ))
+}
+
+fn change_profile_image(image_state: &UseState<String>) {
+    let path = match FileDialog::new().add_filter("image", &["jpg", "png", "jpeg", "svg"]).set_directory(".").pick_file() {
+        Some(path) => path,
+        None => return
+    };
+
+    let file = match std::fs::read(&path) {
+        Ok(image_vec) => image_vec,
+        Err(_) => vec![],
+    };
+
+    let filename = std::path::Path::new(&path)
```

Comment:
No need to convert `path` to `Path` since `path` should be a `PathBuf`. Furthermore this can be simplified by doing

```rust
let filename = path.file_name().map(|file| file.to_string_lossy().to_string()).unwrap_or_default();
```

Although i would probably suggest returning an error to the function itself.